internal/proposals/logs/dynamic-backend-configuration/custom-operator/dynamic-fluentbit-config.md

```diff
@@ -0,0 +1,148 @@
+
+If users want to use more than one pipeline for the log processing, they can use the 'rewrite_tag' filter on their pipeline to create more pipelines. Or they can configure an output plugin to process them with another log processor, as mentioned before.
+
+Additionally, when creating a `CR` a [webhook](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html) is used to validate the correctness of the Fluent Bit configuration based on the Fluent Bit `dry-run` feature.
+
+To actually apply the changes of the users, the operator creates/adapts the ConfigMap for FluentBit and restart Fluent Bit by deleting the pods of the Fluent Bit deployment.
+
+To make sure, that the configuration by the users won't be overwritten by the Reconciler, the basic configuration (`kubernetes` filter, `rewrite_tag`, etc.) is written into a ConfigMap. This ConfigMap is embedded by an `@INCLUDE` statement in the chart of this operator.
+
+<details>
+  <summary><b>CRD Definition (go code) </b>- Click to expand</summary>
+
+```go
+// ConfigSectionSpec defines the desired state of ConfigSection
+type ConfigSectionSpec struct {
+	Type string `json:"type,omitempty"`
+	Entries map[string]string `json:"entries,omitempty"`
+}
+// ConfigSectionStatus defines the observed state of ConfigSection
+type ConfigSectionStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+// ConfigSection is the Schema for the configsections API
+type ConfigSection struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec   ConfigSectionSpec   `json:"spec,omitempty"`
+	Status ConfigSectionStatus `json:"status,omitempty"`
+}
+//+kubebuilder:object:root=true
+// ConfigSectionList contains a list of ConfigSection
+type ConfigSectionList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ConfigSection `json:"items"`
+}
+```
+</details>
+
+
+<details>
+  <summary><b>CRD Definition (yaml file, created using kubebuilder SDK)</b> - Click to expand</summary>
+
+```yaml
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.1
+  creationTimestamp: null
+  name: configsections.logging.kyma-project.io
+spec:
+  group: logging.kyma-project.io
+  names:
+    kind: ConfigSection
+    listKind: ConfigSectionList
+    plural: configsections
+    singular: configsection
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: ConfigSection is the Schema for the configsections API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
```

Comment:
why is schema not capitalized here?

Reply:
Because this references the `schema` in line 136. This code is autogenerated by the SDK and cannot be changed